src/components/MaterialUI/DataDisplay/TableCell.js

```diff
@@ -0,0 +1,54 @@
+import React, { useEffect, useRef } from "react";
+import TableCellMui from "@material-ui/core/TableCell";
+import { makeStyles } from "@material-ui/core/styles";
+import classNames from "classnames";
+import Tooltip from "./Tooltip";
+
+const useStyles = makeStyles(theme => ({
+  cellRoot: {
+    maxWidth: theme.spacing(15),
+    padding: props => theme.spacing(props.cellPaddingInPixels / 10)
+  }
+}));
+
+const TableCell = ({ cell }) => {
+  const cellPaddingInPixels = 16;
+
+  const classes = useStyles({ cellPaddingInPixels });
+
+  const ref = useRef(null);
+
+  const [clientWidth, setWidth] = React.useState(0);
+
+  useEffect(() => {
+    setWidth(ref.current.clientWidth - 2 * cellPaddingInPixels); // left and right padding
+  }, []);
+
+  return (
+    <TableCellMui
+      ref={ref}
+      classes={{
+        root: classNames(classes.cellRoot),
+      }}
+    >
+      {shouldWrapComponentInTooltip(cell) === true ? <Tooltip value={cell} maxWidth={clientWidth} /> : cell}
+    </TableCellMui>
+  );
+};
+
+export const shouldWrapComponentInTooltip = function (component) {
```

Comment:
I thought you would move and use that method in Tooltip component ?

Reply:
Yep we discussed it but, when I think once more I decided that it's not a general behavior. I mean we wan't to wrap cells in Tooltip if they are primitives by default. But, if something was passed in Tooltip component we have to make it tooltip despite on what that component is (if client need to wrap it in tooltip then it should be wrapped). And anyway even primitive should be wrapped in div because of behavior of block element.